.github/workflows/k8s-setup-azure.yml

```diff
@@ -0,0 +1,130 @@
+name: K8s Azure Base Setup
+
+on:
+  workflow_call:
+    inputs:
+      load_balancer_ip:
+        description: "IP Address for the load balancer"
+        type: string
+        required: true
+      load_balancer_id:
+        description: "Resource group where the Public IP is located"
+        required: true
+        type: string
+      issuer_yaml:
+        description: "Yaml file describing cluster issuers"
+        type: string
+        required: true
+      storage_class_yaml:
+        description: "Yaml file describing storage class"
+        type: string
+        required: true
+      cluster_policy_yaml:
+        description: "Yaml file describing the Kyverno cluster policy"
+        type: string
+        required: true
+      git-lfs:
+        description: "Checkout the project with git lfs"
+        type: boolean
+        default: false
+  workflow_dispatch:
+
+jobs:
+  deploy-ingress-controller:
+    runs-on: [ self-hosted, docker ]
+    if: contains('["acpoppe","KlausNie","kupeliorhun"]', github.triggering_actor)
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          lfs: ${{ inputs.git-lfs }}
+      - uses: azure/setup-kubectl@v3
+      - uses: azure/setup-helm@v3
+        with:
+          version: "3.13.3"
+
+      - uses: azure/k8s-set-context@v3
+        with:
+          method: kubeconfig
+          kubeconfig: ${{ secrets.KUBE_CONFIG }}
+          context: ${{ secrets.KUBE_CONTEXT }}
+      - name: Helm Install Ingress Controller
+        run: |
+          helm repo add ingress-nginx https://kubernetes.github.io/ingress-nginx
+          
+          helm repo update
+          
+          helm install ingress-nginx/ingress-nginx \
+            --namespace nginx-ingress-controller \
+            --create-namespace --generate-name \
+            --set controller.service.annotations."service\.beta\.kubernetes\.io/azure-load-balancer-resource-group"="${{ inputs.load_balancer_resource_group }}" \
+            --set controller.service.annotations."service\.beta\.kubernetes\.io/azure-public-ip-address-name"="${{ inputs.public_ip_name }}" \
+            --set controller.service.externalTrafficPolicy="Local" \
+            --set controller.config.enable-ocsp="true" \
+            --set controller.extraArgs.enable-ssl-passthrough="true"
+
+  deploy-cert-manager:
+    runs-on: [ self-hosted, docker ]
+    if: contains('["acpoppe","KlausNie,"kupeliorhun"]', github.triggering_actor)
+    needs: deploy-ingress-controller
+    steps:
+      - name: Wait for Ingress Controller to be ready
+        run: |
+          kubectl wait \
+            --namespace nginx-ingress-controller \
+            --for=condition=ready pod \
+            --selector=app.kubernetes.io/component=controller \
```

Comment:
Not sure we can guarantee this selector is always in place, but if you've checked that it does for the current version of ingress controller, it's fine for now